IOS/SwiftSeedProject/SwiftSeedProject/Infrastructure/PersistenceData.swift

```diff
@@ -20,54 +20,58 @@ class PersistenceData: Persistence {
     }
     
     public func getAll<T: PersistenceObject>(entityName: String) -> [T] {
-        let objectContext = coreDataManager.privateChildManagedObjectContext()
+        let objectContext = coreDataManager.mainManagedObjectContext
         let fetch = NSFetchRequest<NSFetchRequestResult>(entityName: entityName)
         
         do {
             let fetchedObjects = try objectContext.fetch(fetch) as! [T]
             return fetchedObjects
         } catch let error as NSError {
             fatalError("Failed to fetch \(entityName): \(error)")
-            print("Could not getAll. \(error), \(error.userInfo)")
         }
         return []
     }
     
-    public func getBy<T: PersistenceObject>(entityIdentifier: Int, entityName: String) -> T {
-        let objectContext = coreDataManager.privateChildManagedObjectContext()
-        let fetch = NSFetchRequest<NSFetchRequestResult>(entityName: entityName)
-        fetch.predicate = NSPredicate(format: "identifier == %@", entityIdentifier)
+    public func getBy<T: PersistenceObject>(entityIdentifier: String, entityName: String) -> T? {
+        let objectContext = coreDataManager.mainManagedObjectContext
+        let fetch: NSFetchRequest<NSFetchRequestResult> = T.fetchRequest()
+        fetch.predicate = NSPredicate(format: "(identifier = %@)", entityIdentifier)
+        fetch.fetchLimit = 1
         do {
-            let fetchedObjects = try objectContext.fetch(fetch) as! T
-            return fetchedObjects
+            let fetchedObjects = try objectContext.fetch(fetch)
+            return fetchedObjects.count > 0 ? fetchedObjects.item(at: 0) as? T : nil
         } catch let error as NSError {
             fatalError("Failed to fetch \(entityName): \(error)")
-            print("Could not getBy. \(error), \(error.userInfo)")
         }
     }
     
-    public func add(attributes: [String : AnyObject], entityName: String) {
-        let objectContext = coreDataManager.privateChildManagedObjectContext()
+    public func add(attributes: [String : AnyObject]? = nil, entityName: String) -> PersistenceObject {
```

Comment:
why didn't you use:
func add<T: PersistenceObject>(attributes: [String : AnyObject]? = nil, entityName: String) -> T
instead of :
func add(attributes: [String : AnyObject]? = nil, entityName: String) -> PersistenceObject